src/store/media-store.ts

```diff
@@ -222,11 +230,66 @@ export const createSettersSlice: StateCreator<
 
 	_handleProgress: (currentTime: number) =>
 		set(state => {
+			let shouldRunTimeAlarm = false;
 			const currentRelativeTime = Math.min(
 				state.endTime,
 				Math.max(0, currentTime - state.startTime),
 			);
+			let newTimeAlarm = {
+				current: state.currentTimeAlarm || 0,
+				next: state.nextTimeAlarm || 0,
+			};
+			const [timeAlarm, timeBeforeAlarm] = [
+				state.timeAlarm,
+				state.timeBeforeAlarm || 0,
+			];
+
+			// Creating and updating `onTimeAlarm`
+			if (timeAlarm && timeAlarm.length > 0) {
+				// Time that was played previously
+				const previousTime = state.currentTime;
+				// Get the initial state
+				const isInitialized =
```

Comment:
Pls improve the comment. I understand that it sets the initial state (`isInitialized` communicates this clearly) but I don't understand why we can conclude it's initialized if the previously played time is the same as the current and the next alarm...